### ISSUE TITLE
Failed to open database

### ISSUE THREAD

**Issue by LeuAlmeida** [AlexCovizzi/vscode-sqlite#119]
I got [this error](https://github.com/AlexCovizzi/vscode-sqlite/issues/71) - that [was fixed here]- (https://github.com/AlexCovizzi/vscode-sqlite/pull/73) in the latest version.

I resolved that downgrading the version to 0.7.1, but the problem returned.

**Error:**
```shell
Failed to open database: Database failed to open: SQLite process failed to start: The argument 'file' cannot be empty. Received ''
```

**Comment by Henriquepb22** [comment]
I got the same error here, and when browsing to find a fix i found this issue https://github.com/AlexCovizzi/vscode-sqlite/issues/83, so the problem is that some linux (ubuntu 20.04 here) cant run sqlite3 explorer without sqlite3 packages installed, so you need to install it, run:
```bash
 sudo apt install sqlite
```
and i think it will fix it (worked for me)

**Comment by joaozarate** [comment]
> I got the same error here, and when browsing to find a fix i found this issue #83, so the problem is that some linux (ubuntu 20.04 here) cant run sqlite3 explorer without sqlite3 packages installed, so you need to install it, run:
> 
> ```shell
>  sudo apt install sqlite
> ```
> 
> and i think it will fix it (worked for me)

Tnks ... it solved my problem

**Comment by IceHower** [comment]
> I got the same error here, and when browsing to find a fix i found this issue #83, so the problem is that some linux (ubuntu 20.04 here) cant run sqlite3 explorer without sqlite3 packages installed, so you need to install it, run:
> 
> ```shell
>  sudo apt install sqlite
> ```
> 
> and i think it will fix it (worked for me)

Thks.. you are my savior

**Comment by AlexCovizzi** [comment]
I'll investigate the issue this weekend.
@LeuAlmeida It's interesting that it worked with version 0.7.1, i'll look into it, maybe at some point I revereted a fix.
@Henriquepb22 @LeuAlmeida @IceHower @joaozarate Are you all on Ubuntu 20.04?

**Comment by IceHower** [comment]
Yes. On ubuntu need to install the sqlite to work with the extension

**Comment by LeuAlmeida** [comment]
Yes, I am @AlexCovizzi 
Anyway, here we've two solutions, A) **downgrade the version to 0.7.1** and B) installing SQLite via `sudo apt install sqlite`.
I opened this Issue because maybe anyone could have some problem also.

**Comment by W3ND31** [comment]
Guys, u don't need this. If u enabled "sqlite.logLevel":"DEBUG" on VsCode JSON Settings, look for the file like "/home/wendel/.vscode/extensions/alexcvzz.vscode-sqlite-0.8.2/bin/sqlite-v3.26.0-linux-x64"

Go to terminal and type "chmod +x /home/wendel/.vscode/extensions/alexcvzz.vscode-sqlite-0.8.2/bin/sqlite-v3.26.0-linux-x64"

And the issue is resolved without installing anything else.

I think that this will happen only on unix SO's.

**Comment by Querela** [comment]
VSCode with Remote-WSL Ubuntu:
> [12:22:04 PM][vscode-sqlite][INFO] Activating extension vscode-sqlite v0.8.2...
> [12:22:04 PM][vscode-sqlite][ERROR] Unable to find a valid SQLite command. Fallback binary is not valid.
> [12:22:04 PM][vscode-sqlite][INFO] Extension activated.

Had to install `sqlite` (`sqlite3`) as mentioned by @Henriquepb22 [here](https://github.com/AlexCovizzi/vscode-sqlite/issues/119#issuecomment-638507236)

**Comment by freiiirer** [comment]
thks brother.it solved my problem

**Comment by kaueburiti** [comment]
> I got the same error here, and when browsing to find a fix i found this issue #83, so the problem is that some linux (ubuntu 20.04 here) cant run sqlite3 explorer without sqlite3 packages installed, so you need to install it, run:
> 
> ```shell
>  sudo apt install sqlite
> ```
> 
> and i think it will fix it (worked for me)

Just remember to restar your VSCode after doing that ;)
But totally solved my problem!

**Comment by MuhammedNihad** [comment]
> I got the same error here, and when browsing to find a fix i found this issue #83, so the problem is that some linux (ubuntu 20.04 here) cant run sqlite3 explorer without sqlite3 packages installed, so you need to install it, run:
> 
> ```shell
>  sudo apt install sqlite
> ```
> 
> and i think it will fix it (worked for me)

**Thanks brother.**

**Comment by Seratooo** [comment]
> I got the same error here, and when browsing to find a fix i found this issue #83, so the problem is that some linux (ubuntu 20.04 here) cant run sqlite3 explorer without sqlite3 packages installed, so you need to install it, run:
> 
> ```shell
>  sudo apt install sqlite
> ```
> 
> and i think it will fix it (worked for me)

Thanks man, today you helped someone else

**Comment by hochsaison** [comment]
@W3ND31 is totally right. This extension bundles its sqlite binary, no need to install anything as this breaks the DevOps concept. You guy should really first read the whole thread before issuing any commands.

**Comment by igormcsouza** [comment]
> I got the same error here, and when browsing to find a fix i found this issue #83, so the problem is that some linux (ubuntu 20.04 here) cant run sqlite3 explorer without sqlite3 packages installed, so you need to install it, run:
> 
> ```shell
>  sudo apt install sqlite
> ```
> 
> and i think it will fix it (worked for me)

PERFECT!

**Comment by consumerpackagedthoughts** [comment]
> 

Try restarting your vscode after performing that. I fixed mine after applying both.

**Comment by consumerpackagedthoughts** [comment]
> But the error still persists for me on Ubuntu 21.04 x64

Try restarting your vscode after performing those changes - I fixed mine that way.

**Comment by smahi** [comment]
> I got the same error here, and when browsing to find a fix i found this issue #83, so the problem is that some linux (ubuntu 20.04 here) cant run sqlite3 explorer without sqlite3 packages installed, so you need to install it, run:
> 
> ```shell
>  sudo apt install sqlite
> ```
> 
> and i think it will fix it (worked for me)
> 
> **Edit:** check [#119 (comment)](https://github.com/AlexCovizzi/vscode-sqlite/issues/119#issuecomment-639948097) for fix without installing any package

Worked for me on `Pop OS LTS 20.04`

**Comment by Raybeth** [comment]
> I got the same error here, and when browsing to find a fix i found this issue #83, so the problem is that some linux (ubuntu 20.04 here) cant run sqlite3 explorer without sqlite3 packages installed, so you need to install it, run:
> 
> ```shell
>  sudo apt install sqlite
> ```
> 
> and i think it will fix it (worked for me)
> 
> **Edit:** check [#119 (comment)](https://github.com/AlexCovizzi/vscode-sqlite/issues/119#issuecomment-639948097) for fix without installing any package

Thaaaanks! This solved my problem

**Comment by bashidagha** [comment]
> I got the same error here, and when browsing to find a fix i found this issue #83, so the problem is that some linux (ubuntu 20.04 here) cant run sqlite3 explorer without sqlite3 packages installed, so you need to install it, run:
> 
> ```shell
>  sudo apt install sqlite
> ```
> 
> and i think it will fix it (worked for me)
> 
> **Edit:** check [#119 (comment)](https://github.com/AlexCovizzi/vscode-sqlite/issues/119#issuecomment-639948097) for fix without installing any package

Maybe need to restart vscode after this installation
